### PR TITLE
Removing unneeded hook

### PR DESCRIPTION
**Pull request by borkweb**
While looking into the WP 4.5 breaking changes surrounding admin pages for tags, I discovered that our one hook to `load-edit-tags.php` didn't do anything of note.

The associated functions attempt to add a `post_type` argument to the term cloud links on the `edit-tags.php` page for our Event post type. I commented out the hook and then rolled back my WP through the 4.x versions and all the way back to 3.9 - the earliest that we claim to support - and the `post_type` argument was still present.

As such...let's deprecate the two functions and get rid of the hook rather than keeping useless code around.

See: https://central.tri.be/issues/45431

### CODE REVIEW

File: src/Tribe/Main.php [hunk 1]
```diff
@@ -4334,8 +4334,10 @@ public function addHelpAdminMenuItem() {
 		 * When the edit-tags.php screen loads, setup filters
 		 * to fix the tagcloud links
 		 *
+		 * @deprecated 4.1.2
 		 */
 		public function prepare_to_fix_tagcloud_links() {
+			_deprecated_function( __METHOD__, '4.1.2' );
```
Comment (barryhughes) [inline src/Tribe/Main.php]: `__CLASS__ . '::' . __METHOD__`? Just a thought, I don't care that much :smile:

File: src/Tribe/Main.php [hunk 2]
```diff
@@ -4355,6 +4359,7 @@ public function prepare_to_fix_tagcloud_links() {
 		 * @return string
 		 */
 		public function add_post_type_to_edit_term_link( $link, $term_id, $taxonomy, $context ) {
+			_deprecated_function( __METHOD__, '4.1.2' );
```
Comment (barryhughes) [inline src/Tribe/Main.php]: Optionally throw in the __CLASS__ here too (if you also do it in `prepare_to_fix_tagcloud_links()` that is).

File: src/Tribe/Main.php (hunk 1 above)
Comment (borkweb) [inline src/Tribe/Main.php]: `__METHOD__` includes the class already, so this one would be: `Tribe__Events__Main::prepare_to_fix_tagcloud_links`

Comment (borkweb) [inline src/Tribe/Main.php]: I think you may be thinking of `__FUNCTION__`?

Comment (barryhughes) [inline src/Tribe/Main.php]: I think you're right!

**Comment by barryhughes** [comment]
Then merge, sir!

**Pull request merged**